46.md

```diff
@@ -1,98 +1,223 @@
-  - params: [`third-party-pubkey`, `nip44-ciphertext`]
-  - result: `plaintext`
-- **ping**
-  - params: []
-  - result: `"pong"`
+The `content` field is a JSON-RPC-like message that is [NIP-04](https://github.com/nostr-protocol/nips/blob/master/04.md) encrypted and has the following structure:
+
+```json
+{
+    "id": <request_id>,
+    "result": <results_string>,
+    "error": <error_string>
+}
+```
+
+- `id` is the request ID that this response is for.
+- `results` is a string of the result of the call (this can be either a string or a JSON stringified object)
+- `error` is an error in string form.
+
+### Auth Challenges
+
+An Auth Challenge is a response that a remote signer can send back when it needs the user to authenticate via other means. This is currently used in the OAuth-like flow enabled by signers like [Nsecbunker](https://github.com/kind-0/nsecbunkerd/). The response `content` object will take the following form:
+
+```json
+{
+    "id": <request_id>,
+    "result": "auth_url",
+    "error": <URL_to_display_to_end_user>
+}
+```
+
+Clients should display (in a popup or new tab) the URL from the `error` field and then subscribe/listen for another response from the remote signer (reusing the same request ID). This event will be sent once the user authenticates in the other window (or will never arrive if the user doesn't authenticate). It's also possible to add a `redirect_uri` url parameter to the auth_url, which is helpful in situations when a client cannot open a new window or tab to display the auth challenge.
+
+#### Example event signing request with auth challenge
+
+![signing-example-with-auth-challenge](https://i.nostr.build/W3aj.png)
+
+## Remote Signer Commands
+
+Remote signers might support additional commands when communicating directly with it. These commands follow the same flow as noted above, the only difference is that when the client sends a request event, the `p`-tag is the pubkey of the remote signer itself and the `content` payload is encrypted to the same remote signer pubkey.
+
+### Methods/Commands
+
+Each of the following are methods that the client sends to the remote signer.
+
+| Command          | Params                                     | Result                               |
+| ---------------- | ------------------------------------------ | ------------------------------------ |
+| `create_account` | `[<username>, <domain>, <optional_email>]` | `<newly_created_remote_user_pubkey>` |
+
+## Appendix
+
+### NIP-05 Login Flow
+
+Clients might choose to present a more familiar login flow, so users can type a NIP-05 address instead of a `bunker://` string.
+
+When the user types a NIP-05 the client:
+
+- Queries the `/.well-known/nostr.json` file from the domain for the NIP-05 address provided to get the user's pubkey (this is the **remote user pubkey**)
+- In the same `/.well-known/nostr.json` file, queries for the `nip46` key to get the relays that the remote signer will be listening on.
```

Comment:
how do you query for the NIP-46 key?

Reply:
it works just like `relays`, you want to get the `nip46.$pubkey` to get the array of relays for this pubkey:

```bash
$ wget -q -O - https://highlighter.com/.well-known/nostr.json | jq .names.pablo
"bab85564b5d198c3d3a9f856a3fe9dc8e2f9ca3952ba6e7dd60bc98030321103"

$ wget -q -O - https://highlighter.com/.well-known/nostr.json | jq .nip46.bab85564b5d198c3d3a9f856a3fe9dc8e2f9ca3952ba6e7dd60bc98030321103
[
  "wss://relay.nsecbunker.com",
  "wss://relay.damus.io"
]
```

Reply:
ohh the nip46 key is the service provider key. ok.

Reply:
oh wait, there is a nip46 field as well. Interesting.

Reply:
@vitorpamplona my comments are lost in the maelstrom, but I propose to change this to put most of it on the handler event.